# 표준 입력과 표준 출력

In [1]:
12345
10
#ipython 기본 기능
#print(12345) #표준 출력 기능을 내장하고 있음

10

In [2]:
print(12345)
10 #정수형 -> int

12345


10

In [3]:
'12345' #문자열 -> str

'12345'

In [4]:
type(12345)

int

In [5]:
type("12345")

str

In [6]:
type('12345')

str

In [7]:
"12345"*3

'123451234512345'

In [8]:
"num = "+"12345..."

'num = 12345...'

In [9]:
#값 입력 받기
name = input("당신의 이름을 입력하세요: ")
age = input("당신의 나이를 입력하세요: ")
# age = int(input("당신의 나이를 입력하세요: "))

당신의 이름을 입력하세요:  신윤서 
당신의 나이를 입력하세요:  15


In [10]:
print(name, age)

신윤서  15


In [11]:
print("나는",name,"이고 나이는", age, "입니다.")

나는 신윤서  이고 나이는 15 입니다.


In [12]:
print("나는 "+name+"이고 나이는 "+age+"입니다.")

나는 신윤서 이고 나이는 15입니다.


In [13]:
age=int(age)

In [14]:
age + 10

25

# 연습

In [15]:
# 자기 이름으로 삼행시 짓기

print("신: 신기하게 오늘")
print("윤: 윤슬도 봤고")
print("서: 서늘한 바람도 불잖아! 완벽!!")



신: 신기하게 오늘
윤: 윤슬도 봤고
서: 서늘한 바람도 불잖아! 완벽!!


In [16]:
first_line = input("신: ")
sec_line = input("윤: ")
thrid_line = input("서: ")

print(first_line, sec_line, thrid_line, sep="\n ")

신:  신기하게 오늘
윤:  윤슬도 봤고
서:  서늘한 바람도 불잖아! 완벽!!


신기하게 오늘
 윤슬도 봤고
 서늘한 바람도 불잖아! 완벽!!


문자열의 시작이 \ -> 이스케이프 문자, 특수문자 \n \\ \t

In [17]:
삼행시 = first_line + "\n"+ sec_line +"\n"+ thrid_line
print(삼행시)

신기하게 오늘
윤슬도 봤고
서늘한 바람도 불잖아! 완벽!!


# 변수

변수는 type safe: 변수를 선언할 때 타입(유형)을 지정하지 않아도 되는 특성   
변수양 = 값

## 변수의 명명규칙
1. 문자, 숫자, 밑줄을 가지고 만든다
2. 숫자로 시작하면 X
3. 대소문자를 구분
4. 특수기호, 공백 X

In [18]:
x = 10
y = 'hello'
z = 3.141592
x,y,z

(10, 'hello', 3.141592)

In [19]:
type(x),type(y),type(z)

(int, str, float)

In [20]:
print(type(x),type(y),type(z))

<class 'int'> <class 'str'> <class 'float'>


In [21]:
a,b,c = 10,20,30
a,b,c

(10, 20, 30)

In [22]:
a =10; b=20; c=30;

In [23]:
a = b= c= 40

In [24]:
a,b,c

(40, 40, 40)

In [25]:
d = True #bool
e = None #nonetype 
d,e

(True, None)

In [26]:
aa = 10
bb = aa
print(aa,bb)
id(aa), id(bb)

10 10


(140728258169560, 140728258169560)

In [27]:
a = 20
print(a,b)

20 40


In [29]:
#의미부여하기 어려운 변수명 > 수식이 아닌 경우는 사용하지 않음

#의미 있는 변수명

#대문자 upper-case 상수에 사용 
PI = 3.141592
MAX_CONNECTIONS = 10 #no. max connection
MIN_VALUE = 0 #min

# NAME_SPACE

변수가 최초로 정의된 영역에 따라 유효범위가 결정됩니다.

- 지역변수: 함수 내에서 선언
- 전역변수: 함수 영역 밖에서 선언한 변수, global 키워드
- 정적변수ㅣ클래스에서 사용하는 변수

In [33]:
#지역변수 
def local_example():
    local_var = 100
    print("여기는 함수의 영역입니다. ", local_var)

local_example()

여기는 함수의 영역입니다.  100


In [38]:
#전역변수
var_test = "나는 전역이다"
print(var_test)

def global_example():
    global var_test
    var_test = 100
    print("여기는 함수의 영역입니다. ", var_test)

global_example()
print(var_test)

나는 전역이다
여기는 함수의 영역입니다.  100
100


In [49]:
# 변수 영역 연습 

counter = 0 #global

def increment_counter():
    #counter = counter +1
    global counter
    counter += 1
    print("counter : ", counter)
    
increment_counter()
increment_counter()
increment_counter()

counter :  1
counter :  2
counter :  3


# 변수 연습

In [51]:
a = 1
b = "python"
c = [1, 2, 3]

a,b,c

(1, 'python', [1, 2, 3])

In [55]:
#02-8
a = [1, 2, 3]
b = a
print(a)
print(b)
a is b

a[1] = 4
print(a)
print(b)

[1, 2, 3]
[1, 2, 3]
[1, 4, 3]
[1, 4, 3]


In [69]:
#초보자를 위한 파이썬 300제
#파이썬 시작하기
#1
print("Hello World")
#2 
print("Mary's cosmetics")
#3
print('신씨가 소리질렀다. "도둑이야".')
#4
print("C:\Windows")
#5
print("안녕하세요. \n만나서\t반갑습니다.")
#6 
print("오늘은", "일요일")
#7 
print("naver", "kakao", "sk", "samsung", sep=";")
#8
print("naver", "kakao", "sk", "samsung", sep="/")
#9
print("first", end=""); print("second")

Hello World
Mary's cosmetics
신씨가 소리질렀다. "도둑이야".
C:\Windows
안녕하세요. 
만나서	반갑습니다.
오늘은 일요일
naver;kakao;sk;samsung
naver/kakao/sk/samsung
firstsecond


<>:9: SyntaxWarning: invalid escape sequence '\W'
<>:9: SyntaxWarning: invalid escape sequence '\W'
C:\Users\Admin\AppData\Local\Temp\ipykernel_6736\1678798798.py:9: SyntaxWarning: invalid escape sequence '\W'
  print("C:\Windows")


In [72]:
#파이썬 변수 
#11
삼성전자 = 50000
총_평가금액 = 삼성전자*10
print(총_평가금액)
#12
시가총액 = 298000000000000
현재가 = 50000
PER = 15.76
print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))
#13
#14
#15
#16
#17              
#18
#19
#20


500000
298000000000000 <class 'int'>
50000 <class 'int'>
15.76 <class 'float'>


# 자료형

- 기본자료형: 숫자형, 문자열형 str,
- 복합 자료형 : 리스트, 셋, 튜플, 딕셔너리

|type|설명|예|
|-------|-------|-------|
|정수형 (int)| 정수를 저장하는 자료형 |1,2,-5|
|실수형 (float)|소수점이 있는 숫자를 저장하는 자료형|3.14, -0.001, 2.0|
|문자열 (str)|텍스트를 저장하는 자료형|"Hello", '파이썬'|
|논리형 (bool)|참/거짓 값을 저장하는 자료형|True, False|

In [74]:
#수치형 - int
20, type(20)

(20, int)

In [75]:
age = 10
type(age)

int

1. 진법: 10진수, 2진수, 8진수, 16진수
2. 입력하는 방법은 다르지만 print결과는 동일

In [80]:
a=10
b=0b0101 #2wlstn - 0/1fh vygusgksms qkdqjq -> 2^2+2^0
a,b

(10, 5)

In [81]:
c=0o10
d=0b1000
c,d

(8, 8)

In [82]:
#진수 리뷰

In [83]:
#float
b=3.15
b, type(b)

(3.15, float)

지수의 표기

숫자e2 -> 10^2의 곱 = 10**2
숫자e-2 -> 10^-2의 곱 = 10**-2

In [89]:
fa = 10e2
fa, 10**3

(1000.0, 1000)

In [88]:
fb = 10e-2
fb

0.1

In [90]:
fc = 3.1415e9
fc

3141500000.0

In [91]:
#boolean - True, False
#논리연산, 비교연산(T/F)에서 사용

In [93]:
ba = True
#bb = true -> 지정이 안되어있어서

In [94]:
# 논리연산자 - and or not

bc = 1
ba and bc

1

In [95]:
#사칙연산 - + - * / // %
a=2
b=3

In [98]:
a+b
a-b
a*b
a/b
a//b, a%b #몫, remainder


(0, 2)

In [106]:
#비교연산자
1+3 == 4
1+2 != 3
1+3 > 3
1+3 <= 4

True

In [108]:
#비교연산자와 논리연산자의 결합
10%2 == 0 and 12%3==0

True

In [110]:
a = '1234'
b = '5678'
a,b, id(a), id(b)
id(a) == id(b)

False

In [113]:
# 대입연산자
a = 10
b = a+2
a,b
a += 1
a

11

In [116]:
b= 34
b += 10 #44
b -= 5 #39
b

39

# 문자열 str

' ', " ", """ """, ''' ''' docstring

In [117]:
a = "hello"
b = "python"
a+b

'hellopython'

In [118]:
c="*"
c*10

'**********'

In [119]:
e = '1'
e == '1'


True

In [127]:
#문자열은 리스트, 열거형
a = '20010331Rabcd' #list of 13 letters
a[0] #자릿수 ->index
a[3] #indexing
a[0:4] #영역을 지정할 때 : (콜론 사용) -> slicing
a[:4]
a[-5:-2] #a[8:11]
a[9:]

'abcd'

In [129]:
s = 'abc'
a = s
a is s

True

In [132]:
b = s[:]
b
a is b

True